### PR TITLE
[android] fix the build

### PR DESCRIPTION
**Pull request by hyp**
the posix_filesystem module is available only from Swift, as it comes from the Swift android overlay.

### Motivation:

Fixing the android build.

### Modifications:

Modify include.

### Result:

It builds.

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

**Comment by hyp** [comment]
@swift-ci please test

**Review by compnerd — APPROVED** [review]

**Review by stmontgomery — APPROVED** [review]

File: Sources/_TestingInternals/include/Includes.h [hunk 1]
```diff
@@ -140,7 +140,12 @@
 #endif
 
 #if defined(__ANDROID__)
+#ifdef __swift__
```
Comment (grynspan) [inline Sources/_TestingInternals/include/Includes.h:143]:
```suggestion
#if defined(__swift__)
```

File: Sources/_TestingInternals/include/Includes.h [hunk 2]
```diff
@@ -140,7 +140,12 @@
...
+// The linux/stat header is private in the Android modulemap.
 #pragma clang module import posix_filesystem.linux_stat
+#else
```
Comment (grynspan) [inline Sources/_TestingInternals/include/Includes.h:146]: We don't need this `#else` clause.

**Review by grynspan — CHANGES_REQUESTED** [review]
Nitpicks.

**Comment by hyp** [comment]
@swift-ci please test

**Review by grynspan — APPROVED** [review]

**Comment by grynspan** [comment]
@hyp Note if you need this fix in Swift 6.0.1, please clone the PR and nominate it using the procedure outlined [here](https://forums.swift.org/t/swift-6-0-release-process/70220).

**Pull request merged**